test/unit/validators/RunValidator.spec.ts

```diff
@@ -0,0 +1,240 @@
+        // TODO
+        // does not throw error for mode missing
+        expect(nextSpy.calledOnce).to.be.true;
+        // nextSpy is true === next() was called
+    });
+
+    it('should throw an error when mode is NOT string', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                mode: 123,
+                stdin: 'INPUT'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+        // TODO
+        expect(nextSpy.calledOnce).to.be.false;
+        // error thrown == " mode" must be one of [sync, callback, poll]
+        // but should've been "mode" must be string
+    });
+
+    it('should throw an error when mode is not one of "callback, poll, sync"', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                mode: 'wrongMode',
+                stdin: 'INPUT'
+            }
+        };
+
+
+        await runValidator.POST(req, res, nextSpy);
+        // TODO
+        expect(nextSpy.calledOnce).to.be.false;
+        expect(sentData.err.message).to.equal('"mode" must be one of [sync, callback, poll]');
+    });
+
+    it('should not throw an error with STDIN missing', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                mode: 'poll'
+            }
+        };
+        await runValidator.POST(req, res, nextSpy);
+
+        expect(nextSpy.calledOnce).to.be.true;
+    });
+
+    it('should throw an error when STDIN in not string', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                mode: 'poll',
+                stdin: 123
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+
+        expect(nextSpy.calledOnce).to.be.false;
+        expect(sentStatus).to.be.equal(400);
+        expect(sentData.err.message).to.equal('"stdin" must be a string');
+    });
+
+    it('should allow STDIN as empty string', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                mode: 'poll',
+                stdin: ''
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+
+        expect(nextSpy.calledOnce).to.be.true;
+    });
+
+    it('should throw an error when timelimit is not integer', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                mode: 'poll',
+                stdin: '',
+                timelimit: '123'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+
+        // TODO fix this. CODE is wrong
```

Comment:
change this test with `timelimie: 'abc'` and see if it works.

Reply:
it worked :) `'123'` is a string too though :'(